src/nvim/lua/vim.lua

```diff
@@ -439,6 +439,22 @@ function vim.notify(msg, log_level, opts) -- luacheck: no unused
   end
 end
 
+local warn_once = (function()
+  local warned = {}
+  return function(msg)
+    if not warned[msg] then
+      vim.api.nvim_echo({{msg, "WarningMsg"}}, true, {})
```

Comment:
Shouldn't we use nvim_notify?

Reply:
I did consider that, but the role of `vim.notify` is to be intercepted/overridden by the user to show system notifications. I'm not sure whether warnings are something people would want showing up as notifications like that. But I'm open to changing it, I don't feel strongly about it.

Reply:
I think using `nvim_notify` will be nicer.

Reply:
`vim.notify` takes a `log_level` and notify-providers can choose how to display warnings specifically, I thought that was kind of the point? How is this warning different to the warnings `vim.notify` is supposed to handle?